src/formatDate.js

```diff
@@ -49,8 +49,45 @@
  * @returns {string}
  */
 
+const YEAR_FORMAT_BIG = 'YYYY';
+const YEAR_FORMAT_SMALL = 'YY';
+const MONTH_FORMAT = 'MM';
+const DAY_FORMAT = 'DD';
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[fromFormat.length - 1]);
+  const yearIndex = fromFormat.indexOf(YEAR_FORMAT_BIG) === -1
+    ? fromFormat.indexOf(YEAR_FORMAT_SMALL)
+    : fromFormat.indexOf(YEAR_FORMAT_BIG);
+  const year = dateParts[yearIndex];
+  const yearShort = year.slice(-2);
+  const formattedDate = [];
+
+  toFormat.forEach((item) => {
+    if (item === YEAR_FORMAT_SMALL) {
+      formattedDate.push(yearShort);
+    }
+
+    if (item === YEAR_FORMAT_BIG) {
+      const formattedYear = Number(yearShort) < 30 ? `20${yearShort}` : `19${yearShort}`;
+
+      formattedDate.push(formattedYear);
+    }
+
+    if (item === DAY_FORMAT) {
+      const dayIndex = fromFormat.indexOf(DAY_FORMAT);
+
+      formattedDate.push(dateParts[dayIndex]);
+    }
+
+    if (item === MONTH_FORMAT) {
+      const monthIndex = fromFormat.indexOf(MONTH_FORMAT);
+
+      formattedDate.push(dateParts[monthIndex]);
+    }
```

Comment:
In all if statements you check the item, so in this case you use switch case instead, it would be more readable.